File: app/src/main/java/com/hkproductions/listme/host/guestlist/GuestListViewModel.java
package com.hkproductions.listme.host.guestlist;

import android.text.format.Time;
import android.util.Log;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import com.hkproductions.listme.host.database.HostData;
import com.hkproductions.listme.host.database.HostDataDao;

import java.util.Calendar;
import java.util.Date;
import java.util.List;

public class GuestListViewModel extends ViewModel {
    private HostDataDao database;
    public LiveData<List<HostData>> data;
    public MutableLiveData<String> liveName = new MutableLiveData<String>();
    public MutableLiveData<Long> liveDate = new MutableLiveData<Long>();
    public MutableLiveData<Long> liveStartTime = new MutableLiveData<Long>();
    public MutableLiveData<Long> liveEndTime= new MutableLiveData<Long>();
    public GuestListViewModel(HostDataDao database){
        this.database = database;
        liveDate.setValue(Calendar.getInstance().getTimeInMillis());
        liveStartTime.setValue(0L);
        liveEndTime.setValue(0L);
        alterList();
    }
    public void alterList(){
        Log.i("GuestListAltering", String.valueOf(liveDate.getValue()));

        data = database.getEntriesByNameAndTime(liveName.getValue(),
                liveDate.getValue() + liveStartTime.getValue(),
                liveDate.getValue() + liveEndTime.getValue());
    }
}

File: app/src/main/java/com/hkproductions/listme/host/guestlist/GuestListViewModel.kt
package com.hkproductions.listme.host.guestlist

import android.util.Log
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.hkproductions.listme.host.database.HostData
import com.hkproductions.listme.host.database.HostDataDao
import kotlinx.coroutines.launch
import java.util.*

class GuestListViewModel(private val database: HostDataDao) : ViewModel() {
    @JvmField
    var data =
        MutableLiveData<List<HostData>?>()

    @JvmField
    var liveName = MutableLiveData<String>()

    @JvmField
    var liveDate = MutableLiveData<Long>()

    @JvmField
    var liveStartTime = MutableLiveData<Long>()

    @JvmField
    var liveEndTime = MutableLiveData<Long>()

    fun alterList() {
        Log.i("GuestListAltering", liveDate.value.toString())

        viewModelScope.launch {
            data.setValue(
                when (liveName.value.isNullOrBlank()) {
                    true -> database.getEntriesByTime(
                        liveStartTime.value!! + liveDate.value!!,
                        liveEndTime.value!! + liveDate.value!!
                    )
                    false -> database.getEntriesByNameAndTime(
                        liveName.value!!,
                        liveStartTime.value!!.plus(liveDate.value!!),
                        liveEndTime.value!!.plus(liveDate.value!!)
                    )
                }
            )

        }
        Log.i("DebugGuestListNoElement", "name: " + liveName.value)
        Log.i(
            "DebugGuestListNoElement",
            "startTime: " + (liveStartTime.value?.plus(liveDate.value!!))
        )
        Log.i("DebugGuestListNoElement", "endTime: " + (liveEndTime.value?.plus(liveDate.value!!)))
        Log.i(
            "DebugGuestListNoElement",
            "data: " + if (data.value != null) data.value.toString() else ""
        )
    }

    init {
        liveDate.value = Calendar.getInstance().timeInMillis
        liveStartTime.value = 0L
        liveEndTime.value = 0L
        alterList()
    }
}